torchci/components/TorchAgentPage.tsx

```diff
@@ -753,233 +990,351 @@ export const TorchAgentPage = () => {
+            </Typography>
+          </Box>
+        ) : (
+          <TorchAgentPageContainer>
+            {showScrollButton && (
+              <Tooltip title="Go to bottom and resume auto-scroll">
+                <ScrollToBottomButton
+                  variant="contained"
+                  color="primary"
+                  onClick={scrollToBottomAndEnable}
+                  aria-label="Scroll to bottom and resume auto-scroll"
+                >
+                  <ArrowDownwardIcon />
+                </ScrollToBottomButton>
+              </Tooltip>
+            )}
+
+            <Typography variant="h4" gutterBottom>
+              TorchAgent
+            </Typography>
+
+            <Box sx={{ display: "flex", justifyContent: "flex-end", mb: 2 }}>
               <Button
-                variant="contained"
-                color="primary"
-                type="submit"
-                disabled={isLoading}
+                variant="outlined"
+                component="a"
+                href={featureRequestUrl}
+                target="_blank"
+                sx={{ mr: 1 }}
               >
-                {isLoading ? "Running..." : "RUN"}
+                Feature Request
               </Button>
-            </Box>
-          </Box>
-        </Box>
-      </QuerySection>
-
-      <ResultsSection>
-        <Box
-          sx={{
-            display: "flex",
-            justifyContent: "space-between",
-            alignItems: "center",
-            mb: 2,
-          }}
-        >
-          <Typography variant="h6">Results</Typography>
-          {parsedResponses.length > 0 &&
-            parsedResponses.some((item) => item.type === "tool_use") && (
               <Button
                 variant="outlined"
-                size="small"
-                onClick={() => {
-                  if (allToolsExpanded) {
-                    setExpandedTools({});
-                    setAllToolsExpanded(false);
-                  } else {
-                    const allExpanded = parsedResponses.reduce(
-                      (acc, _, index) => {
-                        if (parsedResponses[index].type === "tool_use") {
-                          acc[index] = true;
-                        }
-                        return acc;
-                      },
-                      {} as Record<number, boolean>
-                    );
-                    setExpandedTools(allExpanded);
-                    setAllToolsExpanded(true);
-                  }
-                }}
+                color="error"
+                component="a"
+                href={bugReportUrl}
+                target="_blank"
               >
-                {allToolsExpanded ? "Collapse all tools" : "Expand all tools"}
+                Report Bug
               </Button>
+            </Box>
+
+            {/* Show welcome message and query input only for new chats */}
+            {!selectedSession && (
+              <>
+                <Typography
```

Comment:
maybe split up these blocks into separate components? the jsx code is getting pretty long